java/source/infodynamics/measures/continuous/kraskov/TransferEntropyCalculatorMultiVariateKraskov.java

```diff
@@ -116,6 +118,93 @@
 	 */
 	protected Hashtable<String,String> props;
 
+	/**
+	 * Property name for specifying which (if any) auto-embedding method to use.
+	 * Valid values include {@link #AUTO_EMBED_METHOD_RAGWITZ}, {@link #AUTO_EMBED_METHOD_RAGWITZ_DEST_ONLY},
+	 * {@link #AUTO_EMBED_METHOD_MAX_CORR_AIS}, {@link #AUTO_EMBED_METHOD_MAX_CORR_AIS_DEST_ONLY} 
+	 * {@link #AUTO_EMBED_METHOD_MAX_CORR_AIS_AND_TE}, and {@link #AUTO_EMBED_METHOD_NONE}.
+	 * Defaults to {@link #AUTO_EMBED_METHOD_NONE}
+	 */
+	public static final String PROP_AUTO_EMBED_METHOD = "AUTO_EMBED_METHOD";
+	/**
+	 * Valid value for the property {@link #PROP_AUTO_EMBED_METHOD} indicating that
+	 *  no auto embedding should be done (i.e. to use manually supplied parameters)
+	 */
+	public static final String AUTO_EMBED_METHOD_NONE = "NONE";
+	/**
+	 * Valid value for the property {@link #PROP_AUTO_EMBED_METHOD} indicating that
+	 *  the Ragwitz optimisation technique should be used for automatic embedding
+	 *  for both source and destination time-series
+	 */
+	public static final String AUTO_EMBED_METHOD_RAGWITZ = "RAGWITZ";
+	/**
+	 * Valid value for the property {@link #PROP_AUTO_EMBED_METHOD} indicating that
+	 *  the Ragwitz optimisation technique should be used for automatic embedding
+	 *  for the destination time-series only
+	 */
+	public static final String AUTO_EMBED_METHOD_RAGWITZ_DEST_ONLY = "RAGWITZ_DEST_ONLY";
+	/**
+	 * Valid value for the property {@link #PROP_AUTO_EMBED_METHOD} indicating that
+	 *  the automatic embedding should be done by maximising the bias corrected
+	 *  AIS, for both source and destination time series
+	 */
+	public static final String AUTO_EMBED_METHOD_MAX_CORR_AIS = "MAX_CORR_AIS";
+	/**
+	 * Valid value for the property {@link #PROP_AUTO_EMBED_METHOD} indicating that
+	 *  the automatic embedding should be done by maximising the bias corrected
+	 *  AIS for the source and the TE for the destination.
+	 */
+	public static final String AUTO_EMBED_METHOD_MAX_CORR_AIS_AND_TE = "MAX_CORR_AIS_AND_TE";

```

Comment:
The meaning of this embedding method isn't quite clear from the comment. First, "maximising the bias corrected AIS for the **target**" right? Then, "and subsequently maximising the TE over **source** embeddings, given a fixed source-target delay".
I'm also thinking that it would be good to also optimise the delay here (or rather, to have an option for that), but I think we should come back and revisit that in perhaps another Issue.